sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/ArrayConcatOperatorConversion.java

```diff
@@ -34,14 +34,8 @@ public class ArrayConcatOperatorConversion extends DirectOperatorConversion
       .operandTypeChecker(
           OperandTypes.sequence(
               "(array,array)",
-              OperandTypes.or(
-                  OperandTypes.family(SqlTypeFamily.ARRAY),
-                  OperandTypes.family(SqlTypeFamily.STRING)
-              ),
-              OperandTypes.or(
-                  OperandTypes.family(SqlTypeFamily.ARRAY),
-                  OperandTypes.family(SqlTypeFamily.STRING)
-              )
+              OperandTypes.family(SqlTypeFamily.ANY),
+              OperandTypes.family(SqlTypeFamily.ANY)
```

Comment:
why this change, and similar ones in other files? this seems not as good in terms of validation, since these functions likely can't accept just anything?